library/core/src/intrinsics/simd.rs

```diff
@@ -16,20 +16,54 @@ pub unsafe fn simd_insert<T, U>(_x: T, _idx: u32, _val: U) -> T {
     unreachable!()
 }
 
-/// Extracts an element from a vector.
+/// Inserts an element into a vector, returning the updated vector.
 ///
 /// `T` must be a vector with element type `U`.
 ///
+/// If the index is `const`, [`simd_insert`] may emit better assembly.
+///
 /// # Safety
 ///
 /// `idx` must be in-bounds of the vector.
 #[rustc_intrinsic]
 #[rustc_intrinsic_must_be_overridden]
 #[rustc_nounwind]
+#[cfg(not(bootstrap))]
+pub unsafe fn simd_insert_dyn<T, U>(_x: T, _idx: u32, _val: U) -> T {
```

Comment:
So just to be clear, this will work on all targets, right? Even on x86 and ARM it will generate some correct code when `idx` is a runtime value?

Reply:
Yes, it will degrade significantly on x86 and Arm.

Reply:
As long as it degrades safely and with the right semantics, I am happy. :)